stock_restrict_lot/models/stock_rule.py

```diff
@@ -8,3 +8,15 @@ def _get_custom_move_fields(self):
         fields = super()._get_custom_move_fields()
         fields += ["restrict_lot_id"]
         return fields
+
+    def _push_prepare_move_copy_values(self, move_to_copy, new_date):
+        values = super()._push_prepare_move_copy_values(move_to_copy, new_date)
+        if not move_to_copy.restrict_lot_id:
+            chained_moves = (
+                move_to_copy
+                | move_to_copy.get_all_dest_moves()
+                | move_to_copy.get_all_orig_moves()
+            )
+            if chained_moves.restrict_lot_id:
+                values["restrict_lot_id"] = chained_moves.restrict_lot_id[0].id
+        return values
```

Comment:
I think copying the lot from original move is just enough...

```suggestion
        values = super()._push_prepare_move_copy_values(move_to_copy, new_date)
        values["restrict_lot_id"] = move_to_copy.restrict_lot_id.id if move_to_copy.restrict_lot_id else False 
        return values
```

Reply:
indeed